contracts/test/PDPService.t.sol

```diff
@@ -338,31 +366,96 @@ contract PDPServiceProofSetMutateTest is Test {
         assertEq(pdpService.getRootLeafCount(setId, 2), 0);
 
     }
-
-    function testRemoveBadRootDoesntFail() public {
+    
+    function testRemoveFutureRoots() public {
         uint256 setId = pdpService.createProofSet(address(recordKeeper));
         recordAssert.expectRecord(PDPListener.OperationType.CREATE, setId);
         PDPService.RootData[] memory roots = new PDPService.RootData[](1);
         roots[0] = PDPService.RootData(Cids.Cid(abi.encodePacked("test")), 64);
         pdpService.addRoots(setId, roots);
         recordAssert.expectRecord(PDPListener.OperationType.ADD, setId);
-        uint256[] memory toRemove = new uint256[](1);
-        toRemove[0] = 1;
-        pdpService.removeRoots(setId, toRemove);
+        assertEq(true, pdpService.rootChallengable(setId, 0));
+        uint256[] memory toRemove = new uint256[](2);
 
-        toRemove[0] = 0;
-        pdpService.removeRoots(setId, toRemove);
-        recordAssert.expectRecord(PDPListener.OperationType.REMOVE, setId);
-        assertEq(false, pdpService.rootLive(setId, 0));
+        // Scheduling an un-added root for removal should fail
+        toRemove[0] = 0; // current (unchallengeable) root
+        toRemove[1] = 1; // future root
+        vm.expectRevert("Can only schedule removal of existing roots");
+        pdpService.scheduleRemovals(setId, toRemove);
+        recordAssert.expectRecord(PDPListener.OperationType.REMOVE_SCHEDULED, setId);
+        // Actual removal does not fail
+        pdpService.nextProvingPeriod(setId); // root 0 is now challengable
```

Comment:
But was it removable before it was challengable? This test doesn't exercise that.

Reply:
I think it does if you see the case of root id 1 below